evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,218 @@
+                "value": {
+                    "dataset_name": "commensense_qa",
+                    "skill_type": "multiple-choice",
+                    "metric": "accuracy",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "choices": ["choice.text"],
+                        "answer_index": "1",
+                    },
+                },
+            },
+            "extractive_qa": {
+                "summary": "Extractive sample",
+                "description": "for Extractive Dataset",
+                "value": {
+                    "dataset_name": "quoref",
+                    "skill_type": "extractive-qa",
+                    "metric": "squad",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "context": "context",
+                        "answers": ["answers.text"],
+                    },
+                },
+            },
+        },
+    ),
+):
+    # check if the skill-type is not empty
+    # download the dataset and check if the dataset_name skill-type and metric exist on huggingface
+
+    dataset_handler = DatasetHandler
+    dataset_downloaded = dataset_handler.download_dataset(dataset.dataset_name)
+
+    if dataset_downloaded.name != dataset.dataset_name:
+        raise HTTPException(
+            400,
+            f"skill-type of dataset: {dataset.dataset_name} does not exist on the huggingface ",
+        )
+    # check if the skill-type of dataset exist on huggingface
+
+    if dataset_downloaded.skill_type != dataset.skill_type:
+        raise HTTPException(
+            400,
+            f"skill_type of dataset: {dataset.dataset_name} does not exist on the huggingface ",
+        )
+    # check if the metric of dataset exist on huggingface
+    if dataset_downloaded.metric != dataset.metric:
+        raise HTTPException(
+            400,
+            f"metric of dataset: {dataset.dataset_name} does not exist on the huggingface ",
+        )
+
+        # Check if the dataset exist on the database, when yes, the item muss not be added
+    if mongo_client.client.evaluator.datasets.count_documents(
+        {"dataset_name": dataset.dataset_name}
+    ):
+        raise HTTPException(
+            200, f"dataset: {dataset.dataset_name} exist on the database "
+        )
+
+    else:
+
+        mongo_client.client.evaluator.datasets.insert_one(dataset.mongo())
+        # Download the dataset from
+        dataset_handler = DatasetHandler
+        dataset_handler.get_dataset(dataset.dataset_name)
+
+        logger.info(f"Dataset {dataset.dataset_name} has been added to the database.")
+        m = f"Dataset {dataset.dataset_name} has been added to mongo DB!"
+        raise HTTPException(200, m)
+
+
+@router.get("/{dataset_name}", status_code=200)
+async def get_dataset(
```

Comment:
Logik aus dem Endpunkt muss auch ausgelagert werden, damit die an anderen Stellen in Python aufgerufen werden kann und nicht nur per HTTP-Request

Reply:
was meinst Du hier